elementary/clients/dbt/dbt_runner.py

```diff
@@ -172,3 +173,29 @@ def _get_command_env(self):
     def debug(self, quiet: bool = False) -> bool:
         success, _ = self._run_command(command_args=["debug"], quiet=quiet)
         return success
+
+    def ls(self, select: Optional[str] = None) -> list:
+        command_args = ["ls"]
+        if select:
+            command_args.extend(["-s", select])
+        try:
+            success, command_output_string = self._run_command(
+                command_args=command_args, json_logs=True
+            )
+            command_outputs = command_output_string.splitlines()
+            # ls command didn't match nodes.
+            # When no node is matched, ls command returns 2 dicts with warning mesaggase that there are no matches.
+            if (
+                len(command_outputs) == 2
+                and try_load_json(command_outputs[0])
+                and try_load_json(command_outputs[1])
+            ):
+                logger.warning(
+                    f"The selection criterion '{select}' does not match any nodes"
+                )
+                return []
+            # When nodes are matched, ls command returns strings of the node names.
+            else:
+                return command_outputs
```

Comment:
If nodes are matched, the output of dbt will be:
"node1\nnode2\n"
So I don't see why we should return it like that as well.

When there are no nodes the output is always the same and our warning explains the same thing so it is redundant and will never be used IMO.

WDYT?

Reply:
I mean something else :)
It's good that you split the rows, but I don't get why you call `try_load_json` and then return the non-json loaded values - why not return the output of `try_load_json`?

Reply:
Because I only use `try_load_json` to understand that I have no nodes matches.
Those json has no additional information, and has no use.
I prefer keeping it all simple and the same typing wise:
If there are matches, return a list of the matches.
If there are no matches, log an error and return an empty list.

There is no value by returning 2 dicts that has message that said that there are no node matches and then parse it outside the function.

Reply:
Ah OK, I see what you mean
I misunderstood this. Sababa.